src/initialization/InitDistribution.cpp

```diff
@@ -45,23 +45,27 @@ namespace
         amrex::ParticleReal ix, iy, it, ipx, ipy, ipt;
         amrex::RandomEngine rng;
 
-        if (amrex::ParallelDescriptor::IOProcessor()) {
-            x.reserve(npart);
-            y.reserve(npart);
-            t.reserve(npart);
-            px.reserve(npart);
-            py.reserve(npart);
-            pt.reserve(npart);
-
-            for(amrex::Long i = 0; i < npart; ++i) {
-                distr(ix, iy, it, ipx, ipy, ipt, rng);
-                x.push_back(ix);
-                y.push_back(iy);
-                t.push_back(it);
-                px.push_back(ipx);
-                py.push_back(ipy);
-                pt.push_back(ipt);
-            }
+        int myproc = amrex::ParallelDescriptor::MyProc();
```

Comment:
```suggestion
        // Logic: We initialize 1/Nth of particles, independent of their
        // position, per MPI rank. We then measure the distribution's spatial
        // extent, create a grid, resize it to fit the beam, and then
        // redistribute particles so that they reside on the correct MPI rank.
        int myproc = amrex::ParallelDescriptor::MyProc();
```